packages/channels/src/slack/api.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  AckFn,
+  ButtonAction,
+  ReceiverEvent,
+  RespondFn,
+  SlackAction,
+  SlackActionMiddlewareArgs,
+  SlackEventMiddlewareArgs,
+  StaticSelectAction
+} from '@slack/bolt'
+import crypto from 'crypto'
+import { NextFunction, Response } from 'express'
+import rawBody from 'raw-body'
+import tsscmp from 'tsscmp'
+import { URLSearchParams } from 'url'
+import { Endpoint } from '..'
+import { ChannelApi, ChannelApiManager, ChannelApiRequest } from '../base/api'
+import { POSTBACK_PREFIX, SAY_PREFIX } from '../messenger/renderers/carousel'
+import { QUICK_REPLY_PREFIX } from './renderers/choices'
+import { SlackService } from './service'
+
+export class SlackApi extends ChannelApi<SlackService> {
+  async setup(router: ChannelApiManager) {
+    router.post('/slack', this.verifyRequestSignature.bind(this))
+    router.post('/slack', this.handleRequest.bind(this))
+
+    this.service.on('start', this.handleStart.bind(this))
+  }
+
+  private async verifyRequestSignature(req: ChannelApiRequest, res: Response, next: NextFunction) {
+    const signature = req.headers['x-slack-signature'] as string | undefined
+    const requestTimestamp = req.headers['x-slack-request-timestamp'] as string | undefined
+    const contentType = req.headers['content-type'] as string | undefined
+
+    if (!signature || !requestTimestamp) {
+      return res.sendStatus(401)
+    }
+
+    const ts = Number(requestTimestamp)
+    if (isNaN(ts)) {
```

Comment:
The code in this function is mostly copy paste from the code I found in https://github.com/slackapi/bolt-js/blob/main/src/receivers/ExpressReceiver.ts

I had to write a custom receiver that is adapted to being multi tenant.